models/iaf_cond_alpha_mc_kinetics.cpp

```diff
@@ -0,0 +1,875 @@
+    V_.PSConInit_I_[ n ] = 1.0 * numerics::e / P_.tau_synI[ n ];
+    // changes
+    V_.PSConInit_AP_[ n ] = 1.0 * numerics::e / P_.tau_curAP[ n ];
+  }
+
+  V_.RefractoryCounts_ = Time( Time::ms( P_.t_ref ) ).get_steps();
+
+  assert( V_.RefractoryCounts_ >= 0 ); // since t_ref >= 0, this can only fail in error
+}
+
+
+/* ----------------------------------------------------------------
+ * Update and spike handling functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::iaf_cond_alpha_mc_kinetics::update( Time const& origin, const long_t from, const long_t to )
+{
+
+  assert( to >= 0 && ( delay ) from < kernel().connection_builder_manager.get_min_delay() );
+  assert( from < to );
+
+  for ( long_t lag = from; lag < to; ++lag )
+  {
+
+    double t = 0.0;
+
+    // numerical integration with adaptive step size control:
+    // ------------------------------------------------------
+    // gsl_odeiv_evolve_apply performs only a single numerical
+    // integration step, starting from t and bounded by step;
+    // the while-loop ensures integration over the whole simulation
+    // step (0, step] if more than one integration step is needed due
+    // to a small integration step size;
+    // note that (t+IntegrationStep > step) leads to integration over
+    // (t, step] and afterwards setting t to step, but it does not
+    // enforce setting IntegrationStep to step-t; this is of advantage
+    // for a consistent and efficient integration across subsequent
+    // simulation intervals
+    while ( t < B_.step_ )
+    {
+      const int status = gsl_odeiv_evolve_apply( B_.e_,
+        B_.c_,
+        B_.s_,
+        &B_.sys_,             // system of ODE
+        &t,                   // from t
+        B_.step_,             // to t <= step
+        &B_.IntegrationStep_, // integration step size
+        S_.y_ );              // neuronal state
+
+      if ( status != GSL_SUCCESS )
+        throw GSLSolverFailure( get_name(), status );
+    }
+
+    // changes
+    S_.th_ += ( S_.th_ - P_.V_th ) * ( pow( numerics::e, ( -0.1 / P_.tau_Th ) ) - 1.0 );
+    S_.ICa_ = S_.y_[ DIST * State_::STATE_VEC_COMPS + State_::M_CA ]
+      * S_.y_[ DIST * State_::STATE_VEC_COMPS + State_::H_CA ] * P_.G_Ca
+      * ( P_.E_Ca - S_.y_[ DIST * State_::STATE_VEC_COMPS + State_::V_M ] );
+
+    // add incoming spikes at end of interval
+    // exploit here that spike buffers are compartment for compartment,
+    // alternating between excitatory and inhibitory
+    for ( size_t n = 0; n < NCOMP; ++n )
+    {
+      S_.y_[ n * State_::STATE_VEC_COMPS + State_::DG_EXC ] +=
+        B_.spikes_[ 2 * n ].get_value( lag ) * V_.PSConInit_E_[ n ];
+      S_.y_[ n * State_::STATE_VEC_COMPS + State_::DG_INH ] +=
+        B_.spikes_[ 2 * n + 1 ].get_value( lag ) * V_.PSConInit_I_[ n ];
+    }
+
+
+    // changes
+    // spike handling
+    if ( S_.r_ )
+    { // neuron is absolute refractory
+      --S_.r_;
+      if ( S_.r_ == 10 )
```

Comment:
i mean it to be 1s after spike. but on further thoughts, i decide it to be halfway through the refractory period, changed accordingly.